Mage.Sets/src/mage/cards/s/StormOfSouls.java

```diff
@@ -0,0 +1,153 @@
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * @author Alex-Vasile
+ */
+public class StormOfSouls extends CardImpl {
+    public StormOfSouls(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{4}{W}{W}");
+
+        // Return all creature cards from your graveyard to the battlefield.
+        // Each of them is a 1/1 Spirit with flying in addition to its other types.
+        this.getSpellAbility().addEffect(new StormOfSoulsReturnEffect());
+
+        // Exile Storm of Souls.
+        this.getSpellAbility().addEffect(new ExileSpellEffect().concatBy("<br>"));
+    }
+
+    private StormOfSouls(final StormOfSouls card) { super(card); }
+
+    @Override
+    public StormOfSouls copy() { return new StormOfSouls(this); }
+}
+
+class StormOfSoulsReturnEffect extends OneShotEffect {
+    public StormOfSoulsReturnEffect() {
+        super(Outcome.Benefit);
+        staticText = "Return all creature cards from your graveyard to the battlefield." +
+                "Each of them is a 1/1 Spirit with flying in addition to its other types.";
+    }
+
+    private StormOfSoulsReturnEffect(final StormOfSoulsReturnEffect effect) { super(effect); }
+
+    @Override
+    public StormOfSoulsReturnEffect copy() { return new StormOfSoulsReturnEffect(this); }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+
+        if (player == null) { return false; }
+
+        Set<Card> creatureCardsToBeMovedFromGraveyard = player.getGraveyard().getCards(StaticFilters.FILTER_CARD_CREATURE, game);
+        if (creatureCardsToBeMovedFromGraveyard == null) { return false; }
+
+        boolean allCardsMoved;
+        if (creatureCardsToBeMovedFromGraveyard.isEmpty()) {
+            allCardsMoved = true;
+        } else {
+            // Returns true if all cards were moved.
+            allCardsMoved = player.moveCards(creatureCardsToBeMovedFromGraveyard, Zone.BATTLEFIELD, source, game);
+
+            Set<Card> creatureCardsMovedFromGraveyard;
+
+            // Figure out which cards were successfuly moved so that they can be turned into 1/1 Spirits
+            if (!allCardsMoved) {
+                creatureCardsMovedFromGraveyard = new LinkedHashSet<>();
+
+                for (Card card : creatureCardsToBeMovedFromGraveyard) {
+                    if (game.getState().getBattlefield().containsPermanent(card.getId())) {
+                        creatureCardsMovedFromGraveyard.add(card);
+                    }
+                }
+            } else {
+                creatureCardsMovedFromGraveyard = creatureCardsToBeMovedFromGraveyard;
+            }
+
+            // Change the creatures
+            game.addEffect(new StormOfSoulsChangeCreatureEffect(creatureCardsMovedFromGraveyard), source);
```

Comment:
You should use `effect.setTargetPointer(new FixedTargets())` instead of passing a HashSet to the constructor.